### DISCUSSION TITLE
ETCD HA - RKE config file

### DISCUSSION THREAD

**Discussion by qdupuy** [rancher/rke2#1988] [Q&A]
Hello,

Is it necessary to put the etcd configuration in the config.yaml file on each rke2 master node?

**Comment by brandond** [comment]
Yes, all nodes need to be configured.

**Reply by qdupuy** [reply]
Thank you, regarding the 'server' variable how is it organised when you have several master servers?

On the second node, I add all the public IPs of the masters? and the same on the others except the first?

**Reply by brandond** [reply]
No, for this you would use what we refer to in the docs as a stable registration endpoint. Could be a DNS alias, load balancer, etc. Something that sends traffic to active servers. This is only necessary for registration; once a node joins the cluster it keeps a list of server endpoints in cache on disk.

**Reply by qdupuy** [reply]
so I have to set the server variable on all my nodes?

**Reply by brandond** [reply]
All the nodes that you want to have register with the cluster, yes? How else would they find it?

**Reply by qdupuy** [reply]
I currently have 3 master nodes.

The first one does not have the "server" variable, should I put it in? For the other two servers this will be the case

**Reply by brandond** [reply]
No, the first server node starts the cluster. All other nodes must join the cluster via connecting to an existing server.

**Reply by qdupuy** [reply]
Thank you, I have just created a LB.

Small question, on the canal pods I have this error:

```
k -n kube-system logs -f rke2-canal-wm4xd kube-flannel
I1021 09:44:56.166674       1 main.go:212] Could not find valid interface matching netk8s03@eno2: error looking up interface netk8s03@eno2: route ip+net: no such network interface
E1021 09:44:56.166906       1 main.go:236] Failed to find interface to use that matches the interfaces and/or regexes provided
```

My network configuration was done via netplan on ubuntu.

The iface variable of flannel does not take into account the @ ?

**Reply by brandond** [reply]
Apparently not? Have you tried allowing it to find the correct interface automatically, instead of specifying it?
